File: ehairdressers_mobile/lib/models/appointment_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment_insert_request.g.dart';

@JsonSerializable()
class AppointmentInsertRequest {
  @JsonKey(name: 'EmployeeId')
  final int employeeId;

  @JsonKey(name: 'EmployeeName')
  final String employeeName;

  @JsonKey(name: 'UserId')
  final int userId;

  @JsonKey(name: 'Username')
  final String username;

  @JsonKey(name: 'ServiceId')
  final int serviceId;

  @JsonKey(name: 'ServiceName')
  final String serviceName;

  @JsonKey(name: 'AppointmentDate')
  final String appointmentDate;

  @JsonKey(name: 'AppointmentTime')
  final String appointmentTime;

  AppointmentInsertRequest({
    required this.employeeId,
    required this.employeeName,
    required this.userId,
    required this.username,
    required this.serviceId,
    required this.serviceName,
    required this.appointmentDate,
    required this.appointmentTime,
  });

  factory AppointmentInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$AppointmentInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentInsertRequestToJson(this);
}

File: eHairdressers/eHairdressers.Model/Requests/AppointmentInsertRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.Json.Serialization;

namespace eHairdressers.Model.Requests
{
    public class AppointmentInsertRequest
    {
        [JsonPropertyName("employeeId")]
        public int EmployeeId { get; set; }
        
        [JsonPropertyName("employeeName")]
        public string? EmployeeName { get; set; }
        
        [JsonPropertyName("userId")]
        public int UserId { get; set; }
        
        [JsonPropertyName("username")]
        public string? Username { get; set; }
        
        [JsonPropertyName("serviceId")]
        public int ServiceId { get; set; }
        
        [JsonPropertyName("serviceName")]
        public string? ServiceName { get; set; }
        
        [JsonPropertyName("appointmentDate")]
        public DateTime? AppointmentDate { get; set; }
        
        [JsonPropertyName("appointmentTime")]
        public string? AppointmentTime { get; set; }
        
        [JsonPropertyName("comment")]
        public string? Comment { get; set; }
    }

}
